src/cryptonote_core/blockchain.cpp

```diff
@@ -4034,12 +4034,33 @@ bool Blockchain::for_all_txpool_txes(std::function<bool(const crypto::hash&, con
 
 void Blockchain::set_user_options(uint64_t maxthreads, uint64_t blocks_per_sync, blockchain_db_sync_mode sync_mode, bool fast_sync)
 {
+  if (sync_mode == db_defaultsync)
+  {
+    m_db_default_sync = true;
+    sync_mode = db_async;
+  }
   m_db_sync_mode = sync_mode;
   m_fast_sync = fast_sync;
   m_db_blocks_per_sync = blocks_per_sync;
   m_max_prepare_blocks_threads = maxthreads;
 }
 
+void Blockchain::safesyncmode(const bool onoff)
+{
+  /* all of this is no-op'd if the user set a specific
+   * --db-sync-mode at startup.
+   */
+  if (m_db_default_sync)
+  {
+    /* if turning safemode On, or turning off and not already off */
+    if (onoff || m_db_sync_mode == db_nosync)
```

Comment:
Because at the moment, I only see one possible instance of it being turned on, and multiple possible instances of it being turned off. I.e., I believe there are no redundant calls to turn it on.